pkg/manager/member/tikv_scaler.go

```diff
@@ -103,14 +108,30 @@ func (tsd *tikvScaler) ScaleIn(meta metav1.Object, oldSet *apps.StatefulSet, new
 		return fmt.Errorf("tikvScaler.ScaleIn: failed to get pods %s for cluster %s/%s, error: %s", podName, ns, tcName, err)
 	}
 
-	if tsd.deps.CLIConfig.PodWebhookEnabled {
-		setReplicasAndDeleteSlots(newSet, replicas, deleteSlots)
+	tc, _ := meta.(*v1alpha1.TidbCluster)
+	upNumber := 0
+	storeState := ""
+	for _, store := range tc.Status.TiKV.Stores {
+		if store.State == v1alpha1.TiKVStateUp {
+			upNumber++
+		}
+		if store.PodName == podName {
+			storeState = store.State
+		}
+	}
+	if upNumber < MinimalUpStateOfStores {
+		klog.Errorf("the number of stores in Up state of TidbCluster [%s/%s] is %d, less than %d, can't scale in TiKV, podname %s ", meta.GetNamespace(), meta.GetName(), upNumber, MinimalUpStateOfStores, podName)
 		return nil
+	} else if upNumber == MinimalUpStateOfStores {
+		if storeState == v1alpha1.TiKVStateUp {
+			klog.Errorf("can't scale in TiKV of TidbCluster [%s/%s], cause the number of up stores is %d, and the store in Pod %s which is going to be deleted is up too", meta.GetNamespace(), meta.GetName(), MinimalUpStateOfStores, podName)
```

Comment:
Do we need to record the failed event for debugging purposes? @DanielZhangQD

Reply:
Yes, it's better to record an event for TC here.